### PR TITLE
Added ggcr's k8schain to DefaultKeychain

### PR DESCRIPTION
**Pull request by jabrown85**
The `k8schain` returned by `NewNoClient` will allow image authentication paths on public clouds that were previously unsupported without a cred helper.

The `tools/go.sum` changes brought to you by `make tidy`

Resolves: https://github.com/buildpacks/lifecycle/issues/426

Signed-off-by: Jesse Brown <jabrown85@gmail.com>

### CODE REVIEW

**Comment by natalieparellano** [comment]
@jabrown85 do you have an environment that we could use to validate this?

**Comment by jabrown85** [comment]
@natalieparellano I don't have a public cloud k8s that I can test a successful auth with, but I will spin up a local k8s to run a smoke test to make sure a flow I know works still works.

**Comment by jabrown85** [comment]
I can confirm this works with existing auth flows with no additional logs with the changes in 6d8b3b2. I do not have a way to test a node-identity flow.

File: auth/env_keychain.go [hunk 1]
```diff
@@ -23,9 +27,18 @@ func DefaultKeychain(images ...string) (authn.Keychain, error) {
 		return nil, err
 	}
 
+	// note: context is not used in the NewNoClient path
+	// this adds a credential provider-like keychain for public cloud providers
+	klog.SetLogger(logr.Discard())
```
Comment (jabrown85) [inline auth/env_keychain.go:32]: There didn't seem to be a way to go about discarding logs in any other way. `k8schain` uses `github.com/vdemeester/k8s-pkg-credentialprovider/aws ` which uses `klog` and doesn't seem to be any inputs to send in a customized logger per invocation.

**Comment by jabrown85** [comment]
@jromero you were mentioned as having recently been working on tekton. Do you think you could help test this out in a hosted k8s?

**Review by ekcasey — APPROVED** [review]

File: auth/env_keychain.go (hunk 1 above)
Comment (natalieparellano) [inline auth/env_keychain.go:32]:
```suggestion
	klog.SetLogger(logr.Discard())
	// this adds a credential provider-like keychain for public cloud providers
```

Comment (natalieparellano) [reply auth/env_keychain.go:32]: (to make clearer what the comment is referring to)

File: auth/env_keychain.go [hunk 2]
```diff
@@ -23,9 +27,18 @@ func DefaultKeychain(images ...string) (authn.Keychain, error) {
 		return nil, err
 	}
 
+	// note: context is not used in the NewNoClient path
```
Comment (natalieparellano) [inline auth/env_keychain.go:30]:
```suggestion
```

File: auth/env_keychain.go [hunk 3]
```diff
@@ -23,9 +27,18 @@ func DefaultKeychain(images ...string) (authn.Keychain, error) {
...
+	// this adds a credential provider-like keychain for public cloud providers
+	klog.SetLogger(logr.Discard())
+	clusterNodeChain, err := k8schain.NewNoClient(context.TODO())
```
Comment (natalieparellano) [inline auth/env_keychain.go:33]:
```suggestion
	clusterNodeChain, err := k8schain.NewNoClient(context.TODO()) // note: context is not used in the NewNoClient path
```

**Review by natalieparellano — APPROVED** [review]

**Comment by ekcasey** [comment]
Just noticed, you should probably update the doc comment on above `DefaultKeychain` before merging.

**Pull request merged**